### PR TITLE
Add initial implementation of libp2p consensus adapter

### PR DESCRIPTION
**Pull request by Zeegomo**

### CODE REVIEW

File: nomos-services/consensus/src/network/adapters/libp2p.rs [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+struct MessageCache {
+    // This will always contain VIEW_SIZE_LIMIT consecutive entries
+    cache: Arc<Mutex<BTreeMap<View, Messages>>>,
+}
+
+// This is essentially a synchronization for a single consumer/single producer where the producer must be able to
+// buffer messages even if no consumer showed up yet.
+// Lock-free thread safe ring buffer exists but haven't found a good implementation for rust yet so let's just use
+// channels for now.
+// TODO: replace with vec + notify / wait mechanism
```
Comment (youngjoon-lee) [inline nomos-services/consensus/src/network/adapters/libp2p.rs:50]:
```suggestion
```
I think we can remove this TODO now.

**Review by youngjoon-lee — APPROVED** [review]
👍

File: nomos-services/consensus/src/network/adapters/libp2p.rs [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+#[derive(Clone)]
+struct MessageCache {
+    // This will always contain VIEW_SIZE_LIMIT consecutive entries
+    cache: Arc<Mutex<BTreeMap<View, Messages>>>,
+}
+
+// This is essentially a synchronization for a single consumer/single producer where the producer must be able to
+// buffer messages even if no consumer showed up yet.
+// Lock-free thread safe ring buffer exists but haven't found a good implementation for rust yet so let's just use
+// channels for now.
```
Comment (danielSanchezQ) [inline nomos-services/consensus/src/network/adapters/libp2p.rs:49]: Uhm, maybe a good weekend project! 😄

**Review by danielSanchezQ — APPROVED** [review]
Nice step forward! Thanks!

**Review by al8n — APPROVED** [review]

**Pull request merged**